### PR TITLE
GKEStartPodOperatorAsync inherits GKEStartPodOperator from OSS airflow as the deferrable mode has been implemented

### LINKED ISSUE #956: Remove `get_gke_config_file` in `GKEStartPodOperatorAsync`
Currently, we have used get_gke_config_file https://github.com/astronomer/astronomer-providers/pull/954 which uses gcloud-cli.

Remove the dependency on `get_gke_config_file` in `GKEStartPodOperatorAsync`
Currently when I tried removing and inheriting directly from `GKEStartPodOperator` currently unable to connect to Google Kubernetes Pods it's throwing a connection error with a Name or Service not known. Even though SSL certificate and the endpoint are proper, the namespace and pod name is proper its unable to find the pods. 
We need to debug this further and see why we are unable to connect to Google Kubernetes pods and then remove this dependency of `get_gke_config_file`.

image

### PR DESCRIPTION
**Pull request by Lee-W**
Make `GKEStartPodOperatorAsync` inherits `GKEStartPodOperator` from OSS airflow as the deferrable mode has been implemented after 9.0.0's now release. For backward compatibility, we keep the old implementation for users using ` apache-airflow-providers-google` < 9.0.0

Closes #956

### CODE REVIEW

**Review by pankajastro — APPROVED** [review]
LGTM. but we should test the DAG (integration) tests for both versions once.

**Comment by Lee-W** [comment]
## apache-airflow-providers-google==8.11.0
image

## apache-airflow-providers-google==9.0.0 

image

**Review by pankajkoti — APPROVED** [review]
Minor suggestion. Rest looks good to me.

File: astronomer/providers/google/cloud/operators/kubernetes_engine.py [hunk 1]
```diff
@@ -13,150 +13,88 @@
...
 )
-from astronomer.providers.google.cloud import _get_gke_config_file
+from astronomer.providers.google.cloud import google_provider_version_lt_9
 from astronomer.providers.google.cloud.triggers.kubernetes_engine import (
     GKEStartPodTrigger,
 )
 from astronomer.providers.utils.typing_compat import Context
 
+if google_provider_version_lt_9:
+    from airflow.providers.google.cloud.operators.kubernetes_engine import (
```
[the remark this answers is not in the document]
Comment (Lee-W) [reply astronomer/providers/google/cloud/operators/kubernetes_engine.py]: Yep, this is something I missed. Thanks! Just updated it

**Pull request closed without merging**